web/tests/test_pandas_web.py

```diff
@@ -0,0 +1,82 @@
+from unittest.mock import (
+    mock_open,
+    patch,
+)
+
+import pytest
+import requests
+
+from web.pandas_web import Preprocessors
+
+
+class MockResponse:
+    def __init__(self, status_code: int, response: dict):
+        self.status_code = status_code
+        self._resp = response
+
+    def json(self):
+        return self._resp
+
+    @staticmethod
+    def raise_for_status():
+        return
+
+
+@pytest.fixture
+def context() -> dict:
+    return {
+        "main": {"github_repo_url": "pandas-dev/pandas"},
+        "target_path": "test_target_path",
+    }
+
+
+@pytest.fixture(scope="function")
+def mock_response(monkeypatch, request):
+    def mocked_resp(*args, **kwargs):
+        status_code, response = request.param
+        return MockResponse(status_code, response)
+
+    monkeypatch.setattr(requests, "get", mocked_resp)
+
+
+_releases_list = [
+    {
+        "prerelease": False,
+        "published_at": "2024-01-19T03:34:05Z",
+        "tag_name": "v1.5.6",
+        "assets": None,
+    },
+    {
+        "prerelease": False,
+        "published_at": "2023-11-10T19:07:37Z",
+        "tag_name": "v2.1.3",
+        "assets": None,
+    },
+    {
+        "prerelease": False,
+        "published_at": "2023-08-30T13:24:32Z",
+        "tag_name": "v2.1.0",
+        "assets": None,
+    },
+    {
+        "prerelease": False,
+        "published_at": "2023-04-30T13:24:32Z",
+        "tag_name": "v2.0.0",
+        "assets": None,
+    },
+    {
+        "prerelease": True,
+        "published_at": "2023-01-19T03:34:05Z",
+        "tag_name": "v1.5.3xd",
+        "assets": None,
+    },
```

Comment:
Maybe we can add 10.0 to make sure the possible bug commented above is not there.

Reply:
Added